src/models/Event.js

```diff
@@ -39,27 +40,51 @@ const eventSchema = new Schema({
                     validator: (value) => validTime.test(value),
                     message: 'Invalid time format! Example: hh:mm (24h)'
                 }
-                
             }
-        }, { // Additional date and time (if needed)
-            date: { type: Date },
-            startTime: { type: String },
-            endTime: { type: String }
         }]
     },
-    imageUrl: { type: String, required: true },
+    imageUrl: {
+        type: String,
+        validate: {
+            validator: (value) => validString.test(value),
+            message: "Invalid URL, must start with HTTP:///HTTPS://",
+        },
+    },
+    imageData: {
+        data: Buffer,
+        connectType: String
+    },
     contacts: {
         // TODO: Check later for unique COORDS..!
-        coordinates: { lat: { type: String }, long: { type: String } },
+        coordinates: {
+            type: {
+                type: String,
+                default: 'Point'
+            },
+            coordinates: {
+                type: [Number],
+                required: true
+            }
+        },
         city: { type: String, required: true },
         address: { type: String, required: true },
-        phone: { type: String, required: true }, 
+        phone: { type: String, required: true },
         email: { type: String, required: true }
     },
-    category: { type: String, required: true }, // Event category
-    likedCount: { type: Number, default: 0 }, // Count of users who liked the event
-    creator: { type: ObjectId, ref: 'User', required: true }, // User who created the event (you can specify user properties here)
-    winners: [{ name: { type: String } }, { name: { type: String } }, { name: { type: String } }], // Array of event winners (you can specify winner properties here)
+    category: {
+        type: String,
+        required: true,
+        minlength: [2, "Category must be minimum 2 characters long!"],
+        maxlength: [30, "Category must be maximum 30 characters long!"],
+    },
+    // TODO: check if the User has admin role.
+    likes: [{ type: ObjectId, ref: 'User' } || { type: ObjectId, ref: 'Organization' }],
+    creator: { type: ObjectId, ref: 'User', required: true } || { type: ObjectId, ref: 'Organization', required: true },
+    winners: [
+        { name: String, vehicle: String },
```

Comment:
Why 3 objects in the array?

Reply:
cuz winner may be will be 3 ?

Reply:
Cuz?? So likes (two rows above) are going to be only 1 ?